File: dom/bluetooth/tests/marionette/test_bt_existence.js
/* -*- Mode: js2; js2-basic-offset: 2; indent-tabs-mode: nil; -*- */
/* vim: set ft=javascript ts=2 et sw=2 tw=80: */
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

MARIONETTE_TIMEOUT = 10000;

SpecialPowers.addPermission("bluetooth", true, document);

for (var p in SpecialPowers) {
  log(p);
}
var bt = window.navigator.mozBluetooth;
ok(bt, "BT not found");
ok(bt instanceof BluetoothManager, "BT has wrong interface");
finish();

File: dom/bluetooth/tests/marionette/test_bt_existence.py
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

from marionette_test import MarionetteTestCase
import mozdevice
import os

class TestBTExists(MarionetteTestCase):
    def getJS(self, name):
        return open(os.path.abspath(os.path.join(__file__, os.path.pardir, name + ".js")), 'r')
    def testExist(self):
        result = self.marionette.execute_async_script(self.getJS("test_bt_existence").read())
        self.assertEqual(result["failed"], 0)
